src/Microsoft.Data.SqlClient/tests/UnitTests/Microsoft/Data/SqlTypes/SqlJsonTest.cs

```diff
@@ -0,0 +1,157 @@
+        Assert.Equal(value, json.ToString());
+    }
+
+    [Theory]
+    // Non-string key.
+    [InlineData("{key:\"value\"}")]
+    // Invalid value type.
+    [InlineData("{\"key\":value}")]
+    // Missing closing brace.
+    [InlineData("{\"key\":\"value\"")]
+    // Trailing comma.
+    [InlineData("{\"key\":\"value\",}")]
+    // Comment in JSON.
+    [InlineData("// comment {\"key\":\"value\"}")]
+    public void Constructor_String_Invalid(string invalid)
+    {
+        Assert.ThrowsAny<JsonException>(() => new SqlJson(invalid));
+    }
+
+    // Test the constructor that takes a nullable JsonDocument.
+    [Fact]
+    public void Constructor_JsonDocument_Null()
+    {
+        const JsonDocument? doc = null;
+        SqlJson json = new(doc);
+        Assert.True(json.IsNull);
+        Assert.Throws<SqlNullValueException>(() => json.Value);
+        Assert.Null(json.ToString());
+    }
+
+    [Fact]
+    public void Constructor_JsonDocument_NotNull()
+    {
+        JsonDocument doc = GenerateRandomJson();
+        SqlJson json = new(doc);
+        Assert.False(json.IsNull);
+        Assert.Equal(doc.RootElement.GetRawText(), json.Value);
+        Assert.Equal(doc.RootElement.GetRawText(), json.ToString());
+    }
+
+    [Fact]
+    public void Constructor_JsonDocument_Disposed()
+    {
+        JsonDocument doc = GenerateRandomJson();
+        doc.Dispose();
+        Assert.Throws<ObjectDisposedException>(() => new SqlJson(doc));
+    }
+
+    // IsNull, Value, and ToString() are covered by the above tests.
+
+    // Test that the Value can be round-tripped through a JsonDocument.
+    //
+    // JsonElement.DeepEquals() is only available in .NET 9.0 and later.
+    #if NET9_0_OR_GREATER
+    [Fact]
+    public void RoundTrip()
+    {
+        var doc = GenerateRandomJson();
+        SqlJson json = new(doc);
+
+        var outputDocument = JsonDocument.Parse(json.Value);
+        Assert.True(JsonElement.DeepEquals(
+            doc.RootElement, outputDocument.RootElement));
+    }
+    #endif
+
+    #endregion
+
+    #region Helpers
+
+    private static JsonSerializerOptions _jsonOptions = new()
+    {
+        WriteIndented = true
+    };
+
+    static JsonDocument GenerateRandomJson()
+    {
+
+        Random random = new();
+
+        object jsonObject = new
+        {
+            id = random.Next(1, 1000),
+            name = $"Name{random.Next(1, 100)}",
+            isActive = random.Next(0, 2) == 1,
+            createdDate = DateTime.Now.AddDays(-random.Next(1, 100)).ToString("yyyy-MM-ddTHH:mm:ssZ"),
+            scores = new int[] { random.Next(1, 100), random.Next(1, 100), random.Next(1, 100) },
+            details = new
+            {
+                age = random.Next(18, 60),
+                city = $"City{random.Next(1, 100)}"
+            }
+        };
+
+        string jsonString = JsonSerializer.Serialize(jsonObject, _jsonOptions);
+        return JsonDocument.Parse(jsonString);
```

Comment:
Not sure if this is available everywhere:
```suggestion
        return JsonSerializer.SerializeToDocument(jsonObject, _jsonOptions);
```